src/corelibs/U2View/src/ov_msa/MSAEditorSequenceArea.h

```diff
@@ -233,6 +242,17 @@ private slots:
     QAction *reverseComplementAction;
     QAction *reverseAction;
     QAction *complementAction;
+
+    /**
+     * Set of 'marker' objects from the 'master' components that requested Free ordering mode to be ON are responsible for the 'free' mode ordering.
+     * Free mode can be active only if there is at least one 'marker' in the set.
+     *
+     * When the last marker object is removed from the set the ordering automatically switches to the 'Original'.
+     * Example of master components: multiple synchronized phy-tree views that manage the order of MSA.
+     *
+     * MSAEditor can any time reset this set and switch to 'Original' or 'Sequence' mode.
+     */
+    QSet<QObject *> freeModeMasterMarkersSet;
```

Comment:
I think, we can specify this pointer more tough, at least as a pointer to `GObjectView`. `QObject` too much vague.

Reply:
I have no strong opinion today about the type for the 'marker'. For the purpose we have today it can even be void*.

More than that in the next release it may completely be gone: all depends which of the scenarios described in the bug we will select.